### PR TITLE
Refactor cves.AffectedVersions

### PR DESCRIPTION
**Pull request by andrewpollock**
Store commits in a single array like versions are stored.

With PR https://github.com/google/osv.dev/pull/1508 making combine-to-osv use `cves.VersionInfo`, this makes the
struct more compact, as it's grown organically to be messier than it needs to be.

### CODE REVIEW

File: vulnfeeds/cves/versions.go [hunk 1]
```diff
@@ -613,22 +660,37 @@ func Commit(u string) (string, error) {
 	return "", fmt.Errorf("Commit(): unsupported URL: %s", u)
 }
 
-// For URLs referencing commits in supported Git repository hosts, return a GitCommit.
-func extractGitCommit(link string) *GitCommit {
+// For URLs referencing commits in supported Git repository hosts, return an AffectedCommit.
+// TODO(andrewpollock): do something better aligned with the OSV schema definition once https://github.com/ossf/osv-schema/issues/187
+func extractGitCommit(link string, commitType string) (ac AffectedCommit, err error) {
+	if !slices.Contains([]string{"Introduced", "LastAffected", "Limit", "Fixed"}, commitType) {
```
Comment (another-rex) [inline vulnfeeds/cves/versions.go:666]: Let's change all these `commitType` to the same enum used on vulns.go:418 by moving it out of the function.

**Review by another-rex — COMMENTED** [review]
Merging them makes the code much neater!

File: vulnfeeds/cves/versions.go (hunk 1 above)
Comment (andrewpollock) [reply vulnfeeds/cves/versions.go]: Done in https://github.com/google/osv.dev/pull/1510/commits/e397536f57d7cee950e44093ed3b2470d2d78763

File: vulnfeeds/git/versions.go [hunk 2]
```diff
@@ -33,44 +33,69 @@ func fuzzyVersionToCommit(normalizedVersion string, repo string, normalizedTags
...
-		return gc, false
+		return ac, false
 	}
 	if len(candidateTags) == 1 {
-		return cves.GitCommit{
-			Repo:   repo,
-			Commit: normalizedTags[candidateTags[0]].Commit,
-		}, true
+		ac.SetRepo(repo)
+		switch commitType {
```
Comment (another-rex) [inline vulnfeeds/git/versions.go:40]: Can these also be changed to use the enum?

Comment (andrewpollock) [reply vulnfeeds/git/versions.go:40]: Yes, missed them. Done in https://github.com/google/osv.dev/pull/1510/commits/0a69ddfa848b2722f6e019d5c975877d9015ff32

**Review by another-rex — APPROVED** [review]
LGTM!

**Pull request merged**